src/editor/TextSearch.java

```diff
@@ -0,0 +1,77 @@
+package editor;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TextSearch {
+
+    static  ArrayList<Integer> startIndexes = new ArrayList<>();
```

Comment:
It is also good to make these arrays final. It is a common approach to make final fields of classes whenever it is possible in a multithreaded environment.

Reply:
yes, thank you!